identity/src/main/java/com/stripe/android/identity/navigation/ConsentFragment.kt

```diff
@@ -50,50 +50,62 @@ internal class ConsentFragment(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
-    ): View {
-        binding = ConsentFragmentBinding.inflate(inflater, container, false)
-
-        consentViewModel.loadUriIntoImageView(
-            identityViewModel.verificationArgs.brandLogo,
-            binding.merchantLogo
-        )
-
-        return binding.root
+    ) = ComposeView(requireContext()).apply {
+        setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+        setContent {
+            val verificationState by identityViewModel.verificationPage.observeAsState(Resource.loading())
+            ConsentScreen(
+                verificationState = verificationState,
+                onMerchantViewCreated = {
+                    consentViewModel.loadUriIntoImageView(
+                        identityViewModel.verificationArgs.brandLogo,
+                        it
+                    )
```

Comment:
minor - when possible, using method references instead of lambdas to interact to the viewModel helps avoiding unwanted recompositions. Lambdas will generate an extra class that take the viewModel as a property, and viewModel changes might trigger recompositions.

Found this post very interesting, covering this and other topics: https://multithreaded.stitchfix.com/blog/2022/08/05/jetpack-compose-recomposition/

Reply:
Thanks for the heads up, refactored some methods to be method references